### DISCUSSION TITLE
multiple callback urls for applications

### DISCUSSION THREAD

**Discussion by chrisroat** [community/community#4238] [Other Feature Feedback, Questions, & Ideas]
Giving more visibility to a 6 year issue about oauth callback urls, which I am interested in as well: https://github.com/isaacs/github/issues/422 by @ccutrer

"we have a production version of our application, and then a "beta" version that is a snapshot of the production, and refreshed periodically. when using github to login to our app, it won't work with beta because we can only specify a single valid callback url. google, Facebook, linkedin, and others all support multiple callback urls to solve this situation nicely"

**Comment by YoSev** [comment]
Like others, we are running into the same problem.
As a workaround, we use a webserver we point the callback url to, which forwards the reuqest, depending on a path suffix, to the desired url.

Adding additional parameters to a callback url is valid:
Callback
https://exmaple.com/github/callback
also matches
https://example.com/github/callback/local (forwarded to http://localhost/github/callback)
https://example.com/github/callback/prod (forwarded to http://prod.example.com/github/callback)
https://example.com/github/callback/staging (forwarded to http://staging.example.com/github/callback)

**Comment by hyochan** [comment]
I think this is critical if we want to support multiple devices `web`, `iOS`, `Android` and etc.

Especially if we use cross-platform like https://docs.expo.dev/versions/latest/sdk/auth-session

I hope the Github team take this issue seriously

**Comment by sdarwin** [comment]
Also running into this problem.

Just curious... if the "Authorization callback URL" field is really the URL used in callbacks, it makes sense it would be one value. How could it be otherwise.  Yet google and other providers allow multiple callback urls.  Does that mean the real purpose of this variable is more like "_permitted_ callback urls"?  If so, the actual callback URL is provided by the client during the interaction, and not by this variable?  If that is the case, the field should be converted to a list with multiple values and renamed  "list of permitted callback urls".
